### ISSUE TITLE
sql: pgwire behavior for vector types is partially incorrect

### ISSUE THREAD

**Issue by nvanbenschoten** [cockroachdb/cockroach#13434]
Sequelize queries various `pg_catalog` tables to assemble an array of indexes, using the following query:

```
SELECT i.relname                      AS name, 
       ix.indisprimary                AS PRIMARY, 
       ix.indisunique                 AS unique, 
       ix.indkey                      AS indkey, 
       Array_agg(a.attnum)            AS column_indexes, 
       Array_agg(a.attname)           AS column_names, 
       Pg_get_indexdef(ix.indexrelid) AS definition 
FROM   pg_class t, 
       pg_class i, 
       pg_index ix, 
       pg_attribute a 
WHERE  t.oid = ix.indrelid 
       AND i.oid = ix.indexrelid 
       AND a.attrelid = t.oid 
       AND t.relkind = 'r' 
       AND t.relname = 'user' 
GROUP  BY i.relname, 
          ix.indexrelid, 
          ix.indisprimary, 
          ix.indisunique, 
          ix.indkey 
ORDER  BY i.relname; 
```

Currently, the `pg_index.indkey` is being returned as an `int[]` instead of an `int2vector`, and these have different encodings. The former is an array, while the latter is a space-delimited string. As a result, Sequelize fails to parse the field. We should fix `pg_index` to return the correct type.

A straightforward approach to this would be to add a new OID wrapper type around DArray for this specific instance. We'd also need to allow OID Wrappers to use custom wire encodings when necessary. 

On the other hand, it looks like "vector" types are common in pg_catalog tables. For instance, `pg_index` alone has 4 different fields with 2 different vector types. We may want to look into a more general approach to these types, instead of defaulting to arrays for each instance until we find that they break things.

**Comment by nvanbenschoten** [comment]
First discussed in #12207.

**Comment by cuongdo** [comment]
Partly resolved by #13484, at least for `pg_index.indkey`. There are other vector columns that haven't been changed, so I'm leaving this open.

**Comment by jordanlewis** [comment]
This is fixed.
